categorization/bin/llm_categorization.py

```diff
@@ -1,22 +1,121 @@
+from datetime import datetime
 import json
-from typing import Dict, Any
+from typing import Dict, Any, Union
 from langchain_community.chat_models import ChatOllama
 from langchain_core.prompts import PromptTemplate
 
 
+def SexLevel(result_dict: Dict[str, str], r: str, key: str) -> None:
+    value = result_dict[key]
+    var1: str = ""
+    var2: str = ""
+
+    if "1" in value or "2" in value:
+        var1 = "1"
+        var2 = "2"
+    elif "0" in value or "1" in value:
+        var1 = "0"
+        var2 = "1"
+    elif "M" in value or "F" in value:
+        var1 = "M"
+        var2 = "F"
+    elif "m" in value or "f" in value:
+        var1 = "m"
+        var2 = "f"
+    elif "male" in value or "female" in value:
+        var1 = "male"
+        var2 = "female"
+    else:
+        output: Dict[str, Union[str, Dict[str, str]]] = (
+            {}
+        )  # Or any default output as per your requirement
+        print(json.dumps(output))
+
+    output = {
+        "TermURL": "nb:Sex",
+        "Levels": {str(var1): "male", str(var2): "female"},
+    }
+    print(json.dumps(output))
+
+
+def is_integer(s: str) -> bool:
+    try:
+        int(s)
+        return True
+    except ValueError:
+        return False
+
+
+def is_float(s: str) -> bool:
+    try:
+        float(s)
+        return True
+    except ValueError:
+        return False
+
+
+def is_iso8601(s: str) -> bool:
+    try:
+        datetime.fromisoformat(s)
+        return True
+    except ValueError:
+        return False
+
+
+def is_european_decimal(s: str) -> bool:
+    s = s.strip()  # Remove leading and trailing whitespace
+
+    if s.count(",") == 1:
+        if s.index(",") > 0 and s.index(",") < len(s) - 1:
+            return True
+
+    return False
+
+
+# def is_bounded(s:str)->bool: yet to be added
+# def is_years yet to be added
+
+
+def AgeFormat(result_dict: Dict[str, str], r: str, key: str) -> None:
+    value = result_dict[key].strip()  # Ensure no leading/trailing whitespace
+    numbers_list_str = value.split()
+    Age_l = [num_str.strip() for num_str in numbers_list_str]
+    Fvar: str = "Unknown"
+
+    for num_str in Age_l:
+        if is_integer(num_str):
+            Fvar = "IntValue"
+            break
+        elif is_float(num_str):
+            Fvar = "FloatVal"
+            break
+        elif is_iso8601(num_str):
+            Fvar = "ISO"
+            break
+        elif is_european_decimal(num_str):
+            Fvar = "europeanDecimalValue"
+            break
```

Comment:
```suggestion
        if is_integer(num_str):
            Fvar = "integervalue"
            break
        elif is_float(num_str):
            Fvar = "floatvalue"
            break
        elif is_iso8601(num_str):
            Fvar = "iso8601"
            break
        elif is_european_decimal(num_str):
            Fvar = "europeandecimalvalue"
            break
```
just to match the strings needed for correct parsing